### ISSUE TITLE
Update use of GitHub api octokit package

### ISSUE THREAD

**Issue by webstech** [gitgitgadget/gitgitgadget#206]
After running an npm update to migrate to eslint some errors showed up with the octokit package (still v16).

Easily fixed most problems.
`import { Octokit } from "@octokit/rest";`
`    protected client = new Octokit();`

However, `pulls.createCommentReply` has been deprecated.  It now requires a commit.  Looking for guidance on what is needed here.

**Comment by dscho** [comment]
> However, `pulls.createCommentReply` has been deprecated. It now requires a commit. Looking for guidance on what is needed here.

The [only caller](https://github.com/gitgitgadget/gitgitgadget/blob/d0ebbdfb92a419d3bc507a857c98b9fdf6ebd90f/lib/mail-archive-helper.ts#L142-L143) seems to use the `issueCommentId` [obtained from the Git note associated with the Message-ID of the mail which was replied to](https://github.com/gitgitgadget/gitgitgadget/blob/d0ebbdfb92a419d3bc507a857c98b9fdf6ebd90f/lib/mail-archive-helper.ts#L118).

At the same time we obtain `issueCommentId`, we also get the original commit: https://github.com/gitgitgadget/gitgitgadget/blob/d0ebbdfb92a419d3bc507a857c98b9fdf6ebd90f/lib/mail-archive-helper.ts#L119

The only exception is when we record replies to the cover letter. However, the Git notes _do_ have a commit for those: the tip commit.

My suggestion, therefore, would be to introduce a new local variable, say, `commitFromNote`, and set it to `data.originalCommit` and then use it in the call to `addPRCommentReply()`.

**Comment by webstech** [comment]
Maybe 
```
octokit.pulls.createReviewCommentReply({  owner,  repo,  pull_number,  comment_id,  body});
```
is what we are looking for.  This has the same parms as the current code (different names).

**Comment by dscho** [comment]
> Maybe
> 
> ```
> octokit.pulls.createReviewCommentReply({  owner,  repo,  pull_number,  comment_id,  body});
> ```
> 
> is what we are looking for. This has the same parms as the current code (different names).

Oh yes, this is it! Good find, thank you so much for investigating.

**Comment by webstech** [comment]
@dscho This function was added in [16.29](https://github.com/octokit/rest.js/releases/tag/v16.29.0).  I was running with 16.43.1.  The question is how current are you willing to go?  [V17.0](https://github.com/octokit/rest.js/releases/tag/v17.0.0) is only eight days old so maybe too bleeding edge.

**Comment by dscho** [comment]
@webstech I think we should go for v17. Sure, it's only eight days old, but we're using a pretty stable part of the API.

**Comment by webstech** [comment]
I am working on a dev level test case for github-glue and am seeing a problem in my environment doing commits using the git.ts module.  I would like to confirm if it is localized or not.  The commit timestamps in test jobs (using the dugite package) are incorrect.  Dugite is using `git version 2.20.1.windows.1`.  The commits are showing a timestamp of `Date:   Fri Feb 13 23:33:30 2009 +0000`.  The timestamps are not always exactly the same but very close.  Printing the commit shows the timestamp has a consistent start of '123456'.

```
git cat-file -p e628c81293bb4184a673bc2caebf8220bae9bab0
tree c20b2cc9b78e1523c905ddf2221c33626608e9da
parent 429a81ec83590790f5f402c0b482a6ce25bab010
author joe blow <joe@blow.net> 1234567950 +0000
committer joe blow <joe@blow.net> 1234567950 +0000
```

This is not new - I see test repos from last August showing the same problem.

Is this happening in anyone else's test area or the CI jobs?  Sample output from a test run today:
```
gitgitgadget/.test-dir/trash directory.ci-helper-git-rmtpu4 (master)
$ git log -1
commit 084aeff6096121e0ffd02cef04449a659e3b014c (HEAD -> master, pu, next, maint)
Author: Test User <user@example.com>
Date:   Fri Feb 13 23:32:30 2009 +0000

    A
```

Would like to confirm this is not isolated to me before I dig deeper and start tracing into git.  This only seems to be happening with commits done using dugite.

**Comment by dscho** [comment]
I *think* that's on purpose: the idea is to make the tests reproducible by forcing the `GIT_COMMITTER_DATE`/`GIT_AUTHOR_DATE` to repeatable values. The dates should therefore produce the exact same commit hashes, no matter how often, or where, or when, you run the tests.

**Comment by webstech** [comment]
TIL.  I had not noticed that was fixed in test-lib.ts and not optional.  Thanks for clarifying.

**Comment by webstech** [comment]
I have pushed branch octokit-update.  All the GitHubGlue methods have been tested except annotateCommit.  @dscho I would like to open a pull request but the misc-helper changes have not been fully tested (still working on creating an app) and do not quite match the changes in your auth branch.  Do you see any issues with the code?  It is based on the current doc, which may have changed from when you did your prototype.

A follow on change will update GitHubGlue to support an alternate repository besides `git`.  A dev level unit test will allow devs to run GitHubGlue tests against a different repo.

**Comment by dscho** [comment]
Sure, open that PR! We can always discuss things there.

I will probably be offline most of this weekend, could you ping me on Monday if I didn't have a look by then?

**Comment by webstech** [comment]
PR #210 implements this.  

TIL my PC clock was out of sync (gains ~8 seconds per hour).  It caused this error:
`'Expiration time' claim ('exp') is too far in the future` that took some time to track down.  Updated registry to sync time every few hours.

**Comment by dscho** [comment]
This ticket has been addressed via #210.
